### PR TITLE
Use context users api for People List

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-13252
affects: student
release note: none

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1007018&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Core/Core/API/APICourseRequestable.swift [hunk 1]
```diff
@@ -155,3 +155,17 @@ struct PostCourseRequest: APIRequestable {
...
     }
 }
+
+// https://canvas.instructure.com/doc/api/courses.html#method.courses.users
+public struct GetCourseUsersRequest: APIRequestable {
+    public typealias Response = [APIUser]
+    let courseID: String
+
+    public var path: String {
+        return "\(ContextModel(.course, id: courseID))/users"
```
Comment (vanwagonet) [inline Core/Core/API/APICourseRequestable.swift]: don't we need the `.pathComponent`?

File: Core/Core/API/APICourseRequestable.swift [hunk 2]
```diff
@@ -155,3 +155,17 @@ struct PostCourseRequest: APIRequestable {
         return "\(ContextModel(.account, id: accountID).pathComponent)/courses"
     }
 }
+
+// https://canvas.instructure.com/doc/api/courses.html#method.courses.users
+public struct GetCourseUsersRequest: APIRequestable {
```
Comment (vanwagonet) [inline Core/Core/API/APICourseRequestable.swift]: Do we need this if we have the `GetContextUsersRequest`?

Comment (armstrongnate) [reply Core/Core/API/APICourseRequestable.swift]: Ah nope. This was old news. And that explains the bad path. I'll remove it. Thanks.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-13252](https://instructure.atlassian.net/browse/MBL-13252)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.25% | 93.24% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**